packages/c/sshnpd/src/handle_npt_request.c

```diff
@@ -33,7 +33,11 @@ void handle_npt_request(atclient *atclient, pthread_mutex_t *atclient_lock, sshn
   // allocated: envelope
 
   // log envelope
-  atlogger_log(LOGGER_TAG, ATLOGGER_LOGGING_LEVEL_DEBUG, "Received envelope: %s\n", cJSON_Print(envelope));
+  if (atlogger_get_logging_level() >= ATLOGGER_LOGGING_LEVEL_DEBUG) {
+    char *envelope_str = cJSON_Print(envelope);
+    atlogger_log(LOGGER_TAG, ATLOGGER_LOGGING_LEVEL_DEBUG, "Received envelope: %s\n", envelope_str);
+    free(envelope_str);
+  }
```

Comment:
Pre check if the logger will even print the json before attempting to serialize it. Also, free it.